File: problems/1929-concatenation-of-array/python/test_solution.py
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent))

from solution import Solution


def run_tests() -> None:
    s = Solution()
    cases = [
        ([1, 2, 1], [1, 2, 1, 1, 2, 1]),
        ([1, 3, 2, 1], [1, 3, 2, 1, 1, 3, 2, 1]),
    ]
    for i, (nums, expected) in enumerate(cases, 1):
        result = s.getConcatenation(nums)
        assert result == expected, f"Case {i}: got {result}, want {expected}"
    print(f"All {len(cases)} tests passed.")


if __name__ == "__main__":
    run_tests()

File: problems/1929-concatenation-of-array/javascript/test_solution.js
const { getConcatenation } = require('./solution');

const cases = [
    { nums: [1, 2, 1], expected: [1, 2, 1, 1, 2, 1] },
    { nums: [1, 3, 2, 1], expected: [1, 3, 2, 1, 1, 3, 2, 1] },
];

for (let i = 0; i < cases.length; i++) {
    const { nums, expected } = cases[i];
    const result = getConcatenation(nums);
    if (JSON.stringify(result) !== JSON.stringify(expected)) {
        console.error(`Case ${i + 1}: got ${JSON.stringify(result)}, want ${JSON.stringify(expected)}`);
        process.exit(1);
    }
}

console.log(`All ${cases.length} test cases passed!`);
